### ISSUE TITLE
Available Videos on overview page shows error (using 1.7.1)

### ISSUE THREAD

**Issue by smccarthy** [groupon/Selenium-Grid-Extras#64]
I upgraded to 1.7.1 on my grid and noticed this on the overview page :

Available Videos

    Could not append videos because of java.util.ArrayList cannot be cast to java.util.Map error

**Comment by cktech529** [comment]
Just curious if this will be addressed soon? (There hasn't been any release in months. Is this project over?) Thanks,
Cktech

**Comment by smccarthy** [comment]
@cktech529 I don't think this project is "over". I contribute to it, but haven't lately due to the holidays and work related items.

**Comment by cktech529** [comment]
Thanks!

I see you sent a pull request on the fix. Is it in, awaiting a release? (Something I could pull down, build on my own?) Just curious and thanks again.

cktech

**Comment by smccarthy** [comment]
I didn't send in a pull request for this videos issue. I sent in a pull request for browser version detection and also only ask to record videos for nodes (option 1 and 3), not if the machine is only a hub (option 2).

**Comment by cktech529** [comment]
So I'm now able to record videos (although a bunch of them are coming in all black) what I did to avoid the above issue was change the host name of the hub to the IP (and for some reason, videos are recording and I'm no longer getting the above error.)

**Comment by dimacus** [comment]
@cktech529 if you are running Grid Extras as a service on windows it will record a black video. If you log in directly into the user and run it as a batch file on start up the video should record just fine.

Also, if you are using RDP to connect to the machine, everytime you exit the machine the GUI is diabled and the video will be recorded as black. To fix this, use VNC server to access the machine instead of RDP

**Comment by smccarthy** [comment]
The error , Could not append videos because of java.util.ArrayList cannot be cast to java.util.Map error , appeared because the video_output directory was not there. We now create this directory if it doesn't exist. The fix for this is in master, and will be in the next release.

**Comment by smccarthy** [comment]
Also, my workaround for using RDP and that black screen (at least for the screenshots), was to use the following command to "log off" of the session:
tscon rdp-tcp#0 /dest:console

If I use that, instead of just closing the RDP window, the screenshots appear correctly. However, using VNC might be a better solution.

**Comment by charlescapps** [comment]
I just created a brand new Hub by downloading the newest JAR from https://github.com/groupon/Selenium-Grid-Extras/releases (1.7.1-SNAPSHOT).

And, I'm seeing this error on the hub's page (0 nodes are connected yet):

<pre>
Available Videos
Could not append videos because of java.util.ArrayList cannot be cast to java.util.Map error
</pre>

**Comment by smccarthy** [comment]
@charlescapps The fix isn't in a release yet (it will be in the next release). Though the fix is simple. Create a video_output directory where your jar is.

**Comment by charlescapps** [comment]
Thanks!

**Comment by lettrung99** [comment]
Regarding the comment on Video black out if running selenium node on windows as a service.  Is there a work around to make video to work on windows as a service?  If I don't use as a service then how do I go about making use of the reboot feature on X number of test run on a node?  Please let me know if you have any suggestion here.

**Comment by smccarthy** [comment]
You could put a shell script that starts the selenium-grid-extras jar in the startup directory, and have windows auto login on bootup. Does that work?

**Comment by lettrung99** [comment]
Hi Shawn,
I actually tried something similar as well.  I have a Scheduler Task to trigger a batch file that essentially execute the selenium java command.  This task is set to "run when user is log in".
At the same time, I followed this couple links to enable the auto login.  But somehow it doesn't seems to work for me at all.  
1. Auto-login in Windows Server 20082.  http://www.win2008workstation.com/configure-auto-logon

I am on VM windows 2008 R2 SP1.
Then I tried to use the "run whether user is log on or not" off the Scheduler Task, and the reboot does bring up the selenium service, but the video is Black screen.
Any ideas here?
Thanks,
David
